trusted/wallet.cpp

```diff
@@ -215,7 +223,81 @@ void wallet::on_error(const std::string& message_) const
 
 void wallet::update_stored_wallet() const
 {
-    // TODO Save new stuff to file
+    size_t wallet_size = get_wallet_total_size() + _master_password.size() + 1;
+    size_t sealed_size = wallet_size + sizeof(sgx_sealed_data_t);
+
+    std::unique_ptr<uint8_t[]> sealed_data(new uint8_t[sealed_size]);
+
+    std::unique_ptr<uint8_t[]> data(new uint8_t[sealed_size]);
+
+    auto* output_ = data.get();
+    output_ = std::copy(_master_password.begin(), _master_password.end(), output_);
+    output_++;
+    for (const item_t& item : _items)
+    {
+        output_ = std::copy(item.id.begin(), item.id.end(), output_);
+        output_++;
+        output_ = std::copy(item.username.begin(), item.username.end(), output_);
+        output_++;
+        output_ = std::copy(item.password.begin(), item.password.end(), output_);
+        output_++;
+    }
+
+    sgx_status_t sealing_status = seal_wallet(data.get(), wallet_size, (sgx_sealed_data_t*)sealed_data.get(), sealed_size);
+
+    if (sealing_status != SGX_SUCCESS){
+        on_error("[update_stored_wallet] Wallet not sealed");
+    }
+
+    int ret;
+    sgx_status_t stored_status = store_wallet(&ret, sealed_data.get(), sealed_size);
+    if (ret != 0 || stored_status != SGX_SUCCESS){
+        on_error("[update_stored_wallet] Wallet not stored to file");
+    }  
 }
 
+bool wallet::load_stored_wallet()
+{
+    size_t sealed_size;
+    sgx_status_t status = get_file_size(&sealed_size);
+    if (sealed_size == 0)
+    {
+        return false;
+    }
+    
+    size_t wallet_size = sealed_size - sizeof(sgx_sealed_data_t);
+    uint32_t loaded_size = wallet_size;
+    std::unique_ptr<uint8_t[]> sealed_data(new uint8_t[sealed_size]);
+    std::unique_ptr<uint8_t[]> unsealed_data(new uint8_t[wallet_size]);
+ 
+    int ret;
+    bool stored_status = load_wallet(&ret, sealed_data.get(), sealed_size);
+    if (ret != 0 || stored_status != SGX_SUCCESS){
+        on_error("[load_stored_wallet] Wallet not loaded from file");
```

Comment:
I think that `[load_stored_wallet] Failed to load wallet from file` is better error message.